node/actors/consensus/src/replica/state_machine.rs

```diff
@@ -118,6 +120,16 @@ impl StateMachine {
             block_proposal_cache: self.block_proposal_cache.clone(),
         };
 
-        self.storage.put_replica_state(&backup);
+        let store_result = scope::run_blocking!(ctx, |ctx, s| {
+            let backup_future = self.storage.put_replica_state(ctx, &backup);
+            s.spawn(backup_future).join(ctx).block()?;
+            Ok(())
+        });
+        match store_result {
+            Ok(()) => { /* Everything went fine */ }
+            Err(StorageError::Canceled(_)) => tracing::trace!("Storing replica state was canceled"),
+            Err(StorageError::Database(err)) => panic!("Failed storing replica state: {err}"),
+            // ^ We don't know how to recover from DB errors, so panicking is the only option so far.
```

Comment:
nit: the idiomatic thing would be to return an error here, not panic. It is not up to the code which detected an error to assess whether it is recoverable or not. And even if it is not recoverable now, it doesn't mean that it won't be recoverable in the future. In this particular case, if we abstract out the storage it might be possible that the error is transient for example.